extensions/logging/deployment/src/main/java/io/quarkus/logging/deployment/QuarkusLogProcessor.java

```diff
@@ -0,0 +1,169 @@
+package io.quarkus.logging.deployment;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.LocalVariablesSorter;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.gizmo.Gizmo;
+
+public class QuarkusLogProcessor {
+    private static final DotName QUARKUS_LOG_DOTNAME = DotName.createSimple("io.quarkus.logging.Log");
+
+    private static final String QUARKUS_LOG_BINARY_NAME = "io/quarkus/logging/Log";
+
+    private static final String SYNTHETIC_LOGGER_FIELD_NAME = "quarkusSyntheticLogger";
+
+    private static final String JBOSS_LOGGER_BINARY_NAME = "org/jboss/logging/Logger";
+    private static final String JBOSS_LOGGER_DESCRIPTOR = "L" + JBOSS_LOGGER_BINARY_NAME + ";";
+    private static final String GET_LOGGER_DESCRIPTOR = "(Ljava/lang/String;)" + JBOSS_LOGGER_DESCRIPTOR;
+
+    @BuildStep
+    public void process(CombinedIndexBuildItem index, BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        for (ClassInfo clazz : index.getIndex().getKnownUsers(QUARKUS_LOG_DOTNAME)) {
+            String className = clazz.name().toString();
+
+            transformers.produce(new BytecodeTransformerBuildItem(false, className, (ignored, visitor) -> {
+                return new SynthesiseLogger(visitor, className);
+            }, null, false, ClassReader.EXPAND_FRAMES));
+        }
+    }
+
+    private static class SynthesiseLogger extends ClassVisitor {
```

Comment:
Can we get some Javadoc explaining what the visitor does?

Reply:
I would have hoped that the name is enough, but I'll add something :-)